### PR TITLE
fix(gatsby-plugin-typescript): add missing options validations

### LINKED ISSUE #29063: [gatsby-plugin-typescript]: options schema is missing a few options
All options are passed to babel-preset-typescript, which also supports `jsxPragmaFrag`, `allowNamespaces`, `allowDeclareFields`, and `onlyRemoveTypeImports`.

```
11:30:38 AM: warning Warning: there are unknown plugin options for "gatsby-plugin-typescript": allowNamespaces
11:30:38 AM: Please open an issue at ghub.io/gatsby-plugin-typescript if you believe this option is valid.
```

https://github.com/gatsbyjs/gatsby/blob/bd5b5f75d3c95819c7bf6a548e4c4c98aaae5dfe/packages/gatsby-plugin-typescript/src/gatsby-node.js#L3-L7

https://github.com/gatsbyjs/gatsby/blob/bd5b5f75d3c95819c7bf6a548e4c4c98aaae5dfe/packages/gatsby-plugin-typescript/src/gatsby-node.js#L38-L48

### PR DESCRIPTION
**Pull request by hasparus**
## Description

Added missing field validations, as all options are passed to @babel//preset-typescript.

### Documentation

I'm not sure if the docs need to be updated. They already link to Babel docs.
https://www.gatsbyjs.com/plugins/gatsby-plugin-typescript/

## Related Issues

Closes #29063

### CODE REVIEW

File: packages/gatsby-plugin-typescript/src/__tests__/gatsby-node.js [hunk 1]
```diff
@@ -95,6 +95,9 @@ describe(`gatsby-plugin-typescript`, () => {
         isTSX: false,
         jsxPragma: `ReactFunction`,
```
Comment (hasparus) [inline packages/gatsby-plugin-typescript/src/__tests__/gatsby-node.js:96]:
```suggestion
        jsxPragma: `ReactFunction`,
        jsxPragmaFrag: `React.Fragment`,
```

**Comment by hasparus** [comment]
image

I'm not sure how to fix this.

**Comment by hasparus** [comment]
image

This isn't connected to my PR, right?

**Review by LekoArts — APPROVED** [review]
Thanks!

**Comment by hasparus** [comment]
Thanks, LekoArts!

**Pull request merged**